packages/mui-material/src/Tooltip/Tooltip.js

```diff
@@ -687,54 +694,48 @@ const Tooltip = React.forwardRef(function Tooltip(inProps, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
-  const PopperComponent = slots.popper ?? components.Popper ?? TooltipPopper;
-  const TransitionComponent =
-    slots.transition ?? components.Transition ?? TransitionComponentProp ?? Grow;
-  const TooltipComponent = slots.tooltip ?? components.Tooltip ?? TooltipTooltip;
-  const ArrowComponent = slots.arrow ?? components.Arrow ?? TooltipArrow;
-
-  const popperProps = appendOwnerState(
-    PopperComponent,
-    {
-      ...PopperProps,
-      ...(slotProps.popper ?? componentsProps.popper),
-      className: clsx(
-        classes.popper,
-        PopperProps?.className,
-        (slotProps.popper ?? componentsProps.popper)?.className,
-      ),
+  const externalForwardedProps = {
+    slots,
+    slotProps: {
+      arrow: slotProps.arrow ?? componentsProps.arrow,
+      popper: { ...PopperProps, ...(slotProps.popper ?? componentsProps.popper) },
+      tooltip: slotProps.tooltip ?? componentsProps.tooltip,
+      transition: { ...TransitionProps, ...(slotProps.transition ?? componentsProps.transition) },
     },
+  };
+
+  const [PopperSlot, popperSlotProps] = useSlot('popper', {
+    elementType: components.Popper ?? TooltipPopper,
+    externalForwardedProps,
     ownerState,
-  );
+    className: clsx(classes.popper, PopperProps?.className),
+  });
 
-  const transitionProps = appendOwnerState(
-    TransitionComponent,
-    { ...TransitionProps, ...(slotProps.transition ?? componentsProps.transition) },
+  const [TooltipTransitionSlot, transitionSlotProps] = useSlot('transition', {
+    elementType: components.Transition ?? TransitionComponentProp ?? Grow,
```

Comment:
On the other components we've been migrating `components` and `*Component` props like this:

```diff
 const externalForwardedProps = {
-    slots,
+    slots: {
+        transition: components.transition ?? TransitionComponentProp,
+        ...slots,
+    },
     // ...

 // ...

 const [TooltipTransitionSlot, transitionSlotProps] = useSlot('transition', {
-    elementType: components.Transition ?? TransitionComponentProp ?? Grow,
+    elementType: Grow,
     externalForwardedProps,
```

In other words, using the same structure for `slots` and `slotProps` via `externalForwardedProps`. I think this will make it easier to remove the deprecated `components` and `*Component` props in the future, as all of them will be in one place. What do you think?

Same goes for the other slots in this file.

Reply:
Makes sense to me.

Reply:
Done.